# GLOE Pipe testing and issue documentation

Trying to get this thing to work and want to be prepared when talking to
author hopefully soon.

Trying to get to run with the test data (or any data that is)

## Setup

### Moving files

```
ln -s GLOEPipe/pipelines/GLOEseq/* . 
```
and 
```
ln -s GLOEPipe/modules/GLOEseq/essential.vars.groovy .
```
from README.

Needed to change to `GLOE-Pipe` as this is what it is cloned as.

```
ln -s GLOE-Pipe/pipelines/GLOEseq/* . 
ln -s GLOE-Pipe/modules/GLOEseq/essential.vars.groovy .
```

Copied files from `GLOE-Pipe/test_data` to `rawdata`.

Move `targets.txt` file now in `rawdata` to the outer project directory.

### Test run

Ran `bpipe run gloeseq.pipeline.groovy rawdata/*.fastq.gz`

which produced

```
========================================== Error Details ===========================================

bpipe.PipelineError: A script, requested to be loaded from file './GLOEPipe/modules/GLOEseq/essential.vars.groovy', could not be accessed.
        at bpipe.Pipeline.load(Pipeline.groovy:1275)
        at gloeseq.pipeline.groovy.run(gloeseq.pipeline.groovy:3)
        at gloeseq.pipeline.groovy$run.call(Unknown Source)
        at bpipe.Runner.main(Runner.groovy:491)
        at bpipe.Runner9.main(Runner9.java:47)
```

This implies it is looking for a directory called `GLOEPipe` but cloning
the directory produces `GLOE-Pipe` meaning you need to rename it to
work.

So I renamed `GLOE-Pipe` to `GLOEPipe` and reran that same command above and
remade the sys links since they are now broken.

This failed because I haven't done any work with software installation on 
CRICK but the pipeline did start up.

Opened this [issue](https://gitlab.com/GPetrosino/GLOE-Pipe/-/issues/1) on the GitLab page.

## More setup 

Started modifying the `essential.vars.groovy` file.
Also created genomes folder to hold bowtie index files
for hg19.

You can download those with

```
wget https://genome-idx.s3.amazonaws.com/bt/hg19.zip
unzip hg19.zip
```

Also downloaded chromosome sizes for hg19

```
wget http://hgdownload.cse.ucsc.edu/goldenpath/hg19/bigZips/hg19.chrom.sizes
```

Actually ignore the above I think the test data
is from yeast.

## Looking at bowtie mapping and trying to figure out reverse complement stuff

Samtools uses these flags `def SAMTOOLS_VIEW_FLAG1 = "-bhSu"`

- `-b`: output BAM
- `h`: include header in SAM
- `S`: ignored (input format is auto-detected)
- `u`: uncompressed BAM output

I think we actually want to be looking at the
`bam2bedD.module.groovy` type files for this functionality.

```
bam2bedD = {
	doc title: "bam2bedD",
		desc:  "Convert BAM file to bed and the perl code identifies the first base of each read on the opposite strand (direct mode)",
		constraints: "none.",
		bpipe_version: "tested with bpipe 0.9.9.3.slurm",
		author: "Giuseppe Petrosino"
```

Which would look like

```
| N N N | is read

A T G C A T G | A T G C G A G A T |
                T A C G C T C T A

Then T would be returned?
```




